backend/firestore.rules

```diff
@@ -50,7 +50,8 @@ service cloud.firestore {
       }
 
       allow read: if userIsAuthenticated();
-      allow write: if userIsSelf(resource);
+      allow update, delete: if userIsSelf(resource) && userIsSelf(request.resource);
```

Comment:
Can you briefly explain this change?

Reply:
yes, this verifies that the authenticated user's ID matches both the `auth0Id` field in the existing document and the `auth0Id` field that would result from the incoming change.

Looking at this now, though, I believe there is still a mistake in here, because delete commands won't have an incoming change. (We're not currently deleting user records so this was another oversight in my testing.) It's possible this whole thing is overkill anyway; the critical thing is the new change rule, so I can restore this to its previous state if you think that's best.

Reply:
Thanks for the explanation. What if we just remove the `delete` action from here for now and loop back if and when we need explicit user deletion support?